src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,33 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo;
+  onClose: () => void;
+};
+
+export const TodoModal: React.FC<Props> = ({ todo, onClose }) => {
+  const { id, title, completed, userId } = todo;
+
+  const [user, setUser] = useState<User | null>(null);
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    setLoading(true);
+
+    getUser(userId)
+      .then(setUser)
+      .finally(() => setLoading(false));
+  }, [userId]);
```

Comment:
It's better to move the logic inside the `useEffect `hook to a separate function, as it improves code readability, reusability, and makes testing easier. It also helps keep `useEffect `focused on side effects.